### PR TITLE
Fix various cmake issues

### PR DESCRIPTION
**Pull request by thefallentree**
1. Look for zlib and set HAVE_ZLIB
2. Avoid building and installing doxygen if there is no doxgen.

### CODE REVIEW

File: libtelnet.c [hunk 1]
```diff
@@ -437,7 +436,8 @@ static void _negotiate(telnet_t *telnet, unsigned char telopt) {
 			break;
 		case Q_WANTNO_OP:
 			_set_rfc1143(telnet, telopt, Q_US(q), Q_WANTYES);
-			NEGOTIATE_EVENT(telnet, TELNET_EV_DO, telopt);
+			_send_negotiate(telnet, TELNET_DO, telopt);
```
Comment (seanmiddleditch) [inline libtelnet.c:439]: This don't seem to be related to the CMake issues; can you split these out to a separate PR? (Although I believe these are already addressed by another open PR.)

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,11 +1,16 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.10)
```
Comment (seanmiddleditch) [inline CMakeLists.txt:1]: Is this required? I'm tentative to bump the minimum required here without a really good reason.

**Review by seanmiddleditch — CHANGES_REQUESTED** [review]

**Comment by thefallentree** [comment]
I should close this one for now to catch up until everyone else's has been merged.

File: libtelnet.c (hunk 1 above)
Comment (xiaoxiang781216) [reply libtelnet.c:439]: https://github.com/seanmiddleditch/libtelnet/pull/57
I don't know why my PR mix into your PR.
